### PR TITLE
Audio Preprocessing public sample first checkin

### PR DESCRIPTION
**Pull request by janezdu**
A WPF app that will demo MelSpectrogram generation in a WPF app. 

Currently just displays a static image!

image

### CODE REVIEW

File: Samples/SqueezeNetObjectDetection/Desktop/cpp/LearningModelSessionAPITest.cpp [hunk 1]
```diff
@@ -0,0 +1,1111 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include "testPch.h"
```
Comment (smk2007) [inline Samples/SqueezeNetObjectDetection/Desktop/cpp/LearningModelSessionAPITest.cpp:4]: Is this file needed

Comment (janezdu) [reply Samples/SqueezeNetObjectDetection/Desktop/cpp/LearningModelSessionAPITest.cpp:4]: Nope, thought I cleaned up squeezenet.

File: Samples/SqueezeNetObjectDetection/Desktop/cpp/backup.cpp [hunk 2]
```diff
@@ -0,0 +1,31 @@
+﻿#include "pch.h"
+#include "backup.h"
+#if __has_include("backup.g.cpp")
```
Comment (smk2007) [inline Samples/SqueezeNetObjectDetection/Desktop/cpp/backup.cpp:3]: WHats this implementation for?

File: Samples/SqueezeNetObjectDetection/Desktop/cpp/main.cpp [hunk 3]
```diff
@@ -41,178 +43,149 @@ wstring GetModelPath()
...
-    }
+    ms.MelSpectrogramOnThreeToneSignal(batch_size, signal_size, window_size, dft_size,
+        hop_size, n_mel_bins, sampling_rate);
+    printf("done");
+  
 }
+
+
+
+//bool ParseArgs(int argc, char* argv[])
```
Comment (smk2007) [inline Samples/SqueezeNetObjectDetection/Desktop/cpp/main.cpp:67]: pls remove commented code

File: Samples/AudioPreprocessing/AudioPreprocessing/Model/PreprocessModel.cs [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+            get
+            {
+                return melSpecPath;
+            }
+        }
+
+        public PreprocessModel()
+        {
+            repositoryData = GetData();
+            melSpecPath = "C:/Users/t-janedu/source/repos/Windows-Machine-Learning/Samples/AudioPreprocessing/AudioPreprocessing/tmp/mel_spectrogram_file.jpg";
```
Comment (martinb35) [inline Samples/AudioPreprocessing/AudioPreprocessing/Model/PreprocessModel.cs:32]: can this be a relative path instead of something specific to your machine?

File: Samples/AudioPreprocessing/AudioPreprocessing/Model/PreprocessModel.cs [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+using System.Text;
+using System.Threading.Tasks;
+
+namespace AudioPreprocessing.Model
+{
+    public class PreprocessModel
+    { 
+        private List<string> repositoryData;
+        private string melSpecPath;
+        public string ImportantInfo
```
Comment (martinb35) [inline Samples/AudioPreprocessing/AudioPreprocessing/Model/PreprocessModel.cs:14]: is there a more descriptive name than "important info"? what is the info?

File: Samples/AudioPreprocessing/AudioPreprocessing/Model/PreprocessModel.cs [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+                "world"
+            };
+            return repositoryData;
+        }
+
+        /// <summary>
+        /// Concatenate the information from the list into a fully formed sentence.
+        /// </summary>
+        /// <returns>A string</returns>
+        private string ConcatenateData(List<string> dataList)
```
Comment (martinb35) [inline Samples/AudioPreprocessing/AudioPreprocessing/Model/PreprocessModel.cs:53]: I see what this does, but what is it for? Why is hello world in a list of strings instead of just a string?

File: Samples/AudioPreprocessing/AudioPreprocessing/View/PreprocessView.xaml [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+        xmlns:md="https://github.com/fantasticfiasco/mvvm-dialogs"
+        md:DialogServiceViews.IsRegistered="True"   
+        xmlns:local="clr-namespace:AudioPreprocessing.View"
+        xmlns:vm="clr-namespace:AudioPreprocessing.ViewModel"
+        mc:Ignorable="d"
+        Title="PreprocessingView" Height="800" Width="1300">
+    <Window.DataContext>
+        <vm:PreprocessViewModel/>
+    </Window.DataContext>
+    <!--<Grid>
```
Comment (martinb35) [inline Samples/AudioPreprocessing/AudioPreprocessing/View/PreprocessView.xaml:15]: looks like this Grid is ununsed

File: Samples/AudioPreprocessing/AudioPreprocessing/ViewModel/PreprocessViewModel.cs [hunk 8]
```diff
@@ -0,0 +1,79 @@
...
+         
+            var settings = new OpenFileDialogSettings
+            {
+                Title = "This Is The Title",
+                //InitialDirectory = IOPath.GetDirectoryName(Assembly.GetExecutingAssembly().Location),
+                Filter = "Text Documents (*.txt)|*.txt|All Files (*.*)|*.*"
+            };
+
+            bool? success = dialogService.ShowOpenFileDialog(this, settings);
+            if (success == true)
```
Comment (martinb35) [inline Samples/AudioPreprocessing/AudioPreprocessing/ViewModel/PreprocessViewModel.cs:51]: someone can correct me if I'm out of date, but I believe we avoid the "== true" and just use, "if (success)" in cases like this

File: Samples/AudioPreprocessing/AudioPreprocessing/Model/PreprocessModel.cs (hunk 4 above)
Comment (janezdu) [reply Samples/AudioPreprocessing/AudioPreprocessing/Model/PreprocessModel.cs]: Corrected this!

File: Samples/AudioPreprocessing/AudioPreprocessing/Model/PreprocessModel.cs (hunk 5 above)
Comment (janezdu) [reply Samples/AudioPreprocessing/AudioPreprocessing/Model/PreprocessModel.cs]: This is a residue from a tutorial I followed. Cleaned it up.

File: Samples/AudioPreprocessing/AudioPreprocessing/Model/PreprocessModel.cs (hunk 6 above)
Comment (janezdu) [reply Samples/AudioPreprocessing/AudioPreprocessing/Model/PreprocessModel.cs]: Also a residue from a tutorial.

File: Samples/AudioPreprocessing/AudioPreprocessing/View/PreprocessView.xaml (hunk 7 above)
Comment (janezdu) [reply Samples/AudioPreprocessing/AudioPreprocessing/View/PreprocessView.xaml]: Removed

**Review by ryanlai2 — APPROVED** [review]

File: Samples/AudioPreprocessing/AudioPreprocessing/ViewModel/PreprocessViewModel.cs (hunk 8 above)
Comment (janezdu) [reply Samples/AudioPreprocessing/AudioPreprocessing/ViewModel/PreprocessViewModel.cs]: I took this out completely! Will check in the final version of the file chooser soon!
